validation/src/header.rs

```diff
@@ -86,10 +91,33 @@ pub fn validate_header(
     Ok(())
 }
 
+fn block_timestamp_less_than_2h_from_current_time(
+    block_time: u64,
+    current_time: u64,
+) -> Result<(), ValidateHeaderError> {
+    let number_of_seconds_in_one_hour = 60 * 60;
+    let max_allowed_time = current_time + 2 * number_of_seconds_in_one_hour;
+
+    if block_time > max_allowed_time {
+        return Err(ValidateHeaderError::HeaderIsTooFarInFuture {
+            block_time,
+            max_allowed_time,
+        });
+    }
+
+    Ok(())
+}
+
 /// Validates if a header's timestamp is valid.
 /// Bitcoin Protocol Rules wiki https://en.bitcoin.it/wiki/Protocol_rules says,
 /// "Reject if timestamp is the median time of the last 11 blocks or before"
-fn is_timestamp_valid(store: &impl HeaderStore, header: &BlockHeader) -> bool {
+/// "Block timestamp must not be more than two hours in the future"
+fn is_timestamp_valid(
+    store: &impl HeaderStore,
+    header: &BlockHeader,
+    current_time: u64,
+) -> Result<(), ValidateHeaderError> {
+    block_timestamp_less_than_2h_from_current_time(header.time as u64, current_time)?;
```

Comment:
I think this method name is misleading. We're not verifying that the block timestamp is less than 2h from the current time.

Reply:
Changed.